### PR TITLE
Update module versions, examples and tests

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Update module versions, examples and tests
* Update README and LICENSE
* Fix TF code formatting 

## why
* Keep up to date

### CODE REVIEW

File: iam-role.tf [hunk 1]
```diff
@@ -63,7 +65,7 @@ data "aws_iam_policy_document" "ssm" {
 resource "aws_iam_policy" "ssm" {
   count = try((local.enabled && var.ssm_parameter_names != null && length(var.ssm_parameter_names) > 0), false) ? 1 : 0
 
-  description = "Provides minimum SSM read permissions."
+  description = "Lambda policy with minimum SSM read permissions"
```
Comment (nitrocode) [inline iam-role.tf:68]: This will cause a recreation iirc

File: examples/docker-image/main.tf [hunk 2]
```diff
@@ -37,11 +43,11 @@ resource "null_resource" "docker_build" {
 resource "null_resource" "docker_login" {
   count = module.this.enabled ? 1 : 0
   provisioner "local-exec" {
-    command = "aws ecr get-login-password --region ${data.aws_region.this[0].name} | docker login --username AWS --password-stdin ${data.aws_caller_identity.this[0].account_id}.dkr.ecr.${data.aws_region.this[0].name}.amazonaws.com"
+    command = "aws ecr get-login-password --region ${join("", data.aws_region.this.*.name)} | docker login --username AWS --password-stdin ${join("", data.aws_caller_identity.this.*.account_id)}.dkr.ecr.${join("", data.aws_region.this.*.name)}.amazonaws.com"
   }
 
   depends_on = [
-    null_resource.docker_build,
+    null_resource.docker_build
```
[the remark this answers is not in the document]
Comment (Nuru) [reply examples/docker-image/main.tf:50]: I think the style convention we have adopted is that if there is only 1 entry, we do not use a trailing comma (and usually we put the list all on one line), and only leave a trailing comma for a longer list (where a comma is required after the other entries).

File: examples/docker-image/main.tf [hunk 3]
```diff
@@ -53,20 +59,20 @@ resource "null_resource" "docker_push" {
 
   depends_on = [
     time_sleep.wait_15_seconds,
-    null_resource.docker_login,
+    null_resource.docker_login
```
[the remark this answers is not in the document]
Comment (Nuru) [reply examples/docker-image/main.tf:62]: So here I would keep the comma

File: examples/docker-image/main.tf [hunk 4]
```diff
@@ -53,20 +59,20 @@ resource "null_resource" "docker_push" {
...
   source = "../.."
 
   function_name = module.label.id
   image_uri     = "${module.ecr.repository_url}:latest"
   package_type  = "Image"
 
   context = module.this.context
+
   depends_on = [
-    null_resource.docker_push,
```
[the remark this answers is not in the document]
Comment (Nuru) [reply examples/docker-image/main.tf:70]: Here I would leave the comma out

File: test/src/go.mod [hunk 5]
```diff
@@ -1,9 +1,58 @@
 module github.com/cloudposse/terraform-aws-lambda-function
 
-go 1.16
+go 1.18
 
 require (
 	github.com/aws/aws-sdk-go v1.40.56
 	github.com/gruntwork-io/terratest v0.40.0
```
Comment (Nuru) [inline test/src/go.mod:7]: Update `terratest` to `v0.40.16` or later because of known issues with `v0.40.15`or earlier.

File: test/src/go.mod [hunk 6]
```diff
@@ -1,9 +1,58 @@
 module github.com/cloudposse/terraform-aws-lambda-function
 
-go 1.16
+go 1.18
 
 require (
 	github.com/aws/aws-sdk-go v1.40.56
```
Comment (Nuru) [inline test/src/go.mod:6]: Use current v1.44.49 or almost current (survived a week without bug reports or fixes) v1.44.47 or document why you are using an older version.

**Review by Nuru — CHANGES_REQUESTED** [review]
Please update the test's `go` package dependencies to current versions

File: main.tf [hunk 7]
```diff
@@ -10,7 +17,8 @@ module "cloudwatch_log_group" {
 }
 
 resource "aws_lambda_function" "this" {
-  count      = module.this.enabled ? 1 : 0
+  count = module.this.enabled ? 1 : 0
+
   depends_on = [module.cloudwatch_log_group]
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply main.tf:22]: Yes that does look nice

**Review by nitrocode — APPROVED** [review]

**Review by Nuru — APPROVED** [review]

**Pull request merged**